streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimestampedTupleForwarder.java

```diff
@@ -38,7 +39,7 @@
     @SuppressWarnings({"unchecked", "rawtypes"})
     TimestampedTupleForwarder(final StateStore store,
                               final ProcessorContext<K, Change<V>> context,
-                              final TimestampedCacheFlushListener<K, V> flushListener,
+                              final CacheFlushListener<K, ?> flushListener,
```

Comment:
This is 3) in the description: as we can use the base class, we then would not need the duplicated TupleFowarder any more.